### ISSUE TITLE
Pinned items should have a pin icon

### ISSUE THREAD

**Issue by ffMathy** [ffMathy/Shapeshifter#435]
Pinned items should either be a different color (maybe a different theme or chrome) and perhaps an icon.

**Comment by thegabrielstein** [comment]
Maybe add a simple pin icon? Maybe a rectangle around the icon?

![](https://i.imgur.com/IGnp6DB.png)

**Comment by thegabrielstein** [comment]
This is significant if it's a solution to persistent items like in #31 or #459. Can't tell or trust that items are saved or persistent.

**Comment by ffMathy** [comment]
Item that are pinned are stored on disk already and will be on the top always too 😉

**Comment by thegabrielstein** [comment]
Hmm, how do you actually pin though? Seems like a dumb question, but no key actually seems to do anything. Going into the action menu and moving down to that item and then waiting, pressing enter, pressing space, pressing left arrow, releasing Control + V, etc seem to actually pin it? 

What am I missing, and at the very least having a wiki page with ideal workflows. And this is probably a part of #28 or coaching users on startup. 

Seems like Left Arrow key is the most intuitive way to select an action menu item if Right Arrow escapes or closes?

**Comment by ffMathy** [comment]
1. Hold down CTRL V.
2. Hit left arrow key to go to the action list. 
3. Use up or down arrow keys to select an action. 
4. Release CTRL V to perform that action. 

I actually thought this was intuitive, and if you look carefully, the UI alters a bit when hitting the left and right key back and forth a little to indicate which list you are on. 

That's why it's good to have someone who is not a developer of the project leave feedback, because I am clearly biased! 😎

The coaching thing now seems more important than ever and I have an idea of how it could be implemented quite simply without too much effort!

Ideally there should be a tutorial section for every feature in Shapeshifter going forward 👍

Perhaps reprioritize that issue? 😊

**Comment by thegabrielstein** [comment]
Correct, left and right keys are intuitive, especially right arrow to cancel paste. But releasing Control + V isn't intuitive because it seems like you're canceling the action and you don't see any visual confirmation of the action being completed. No popup saying "It's pinned" etc. 

Ideally though left arrow would also select an action menu item which would offer the opportunity for a brief flash animation or a loading bar line to travel through the menu item indicating that it's completed. Or even a small check mark icon appearing over it? 

Could be both at the end of the day. Just a high likelihood that more actions are added that would benefit from the overlay remaining open while performing action menu tasks.

And yeah, definitely a tutorial is high priority for training users on use of the software!
